src/main/surfrdUtilsMod.F90

```diff
@@ -145,6 +146,86 @@ subroutine convert_cft_to_pft( begg, endg, cftsize, wt_cft )
 
   end subroutine convert_cft_to_pft
 
+  !-----------------------------------------------------------------------
+  subroutine collapse_individual_lunits(wt_lunit, begg, endg, toosmall_soil, &
+                                        toosmall_crop, toosmall_glacier, &
+                                        toosmall_lake, toosmall_wetland, &
+                                        toosmall_urb_tbd, toosmall_urb_hd, &
+                                        toosmall_urb_md)
+    ! DESCRIPTION
+    ! Keep landunits above the user-defined thresholds and remove the rest
+    !
+    ! !USES:
+    use landunit_varcon, only: max_lunit, istsoil, istcrop, istice_mec, &
+                               istdlak, istwet, isturb_tbd, isturb_hd, &
+                               isturb_md
+    !
+    ! !ARGUMENTS:
+    integer, intent(in) :: begg  ! Beginning grid cell index
+    integer, intent(in) :: endg  ! Ending grid cell index
+    integer, intent(in) :: toosmall_soil  ! Soil landunit threshold (%)
+    integer, intent(in) :: toosmall_crop  ! Crop landunit threshold (%)
+    integer, intent(in) :: toosmall_glacier  ! Glacier landunit threshold (%)
+    integer, intent(in) :: toosmall_lake  ! Lake landunit threshold (%)
+    integer, intent(in) :: toosmall_wetland  ! Wetland landunit threshold (%)
+    integer, intent(in) :: toosmall_urb_tbd  ! Urban TBD landunit threshold (%)
+    integer, intent(in) :: toosmall_urb_hd  ! Urban HD landunit threshold (%)
+    integer, intent(in) :: toosmall_urb_md  ! Urban MD landunit threshold (%)
+    ! This array modified in-place
+    ! Weights of landunits per grid cell
+    real(r8), intent(inout) :: wt_lunit(begg:endg, max_lunit)
+    !
+    ! !LOCAL VARIABLES:
+    integer :: g  ! grid cell indexes
+    integer :: m  ! landunit indexes
+    integer :: max_landunit  ! landunit with largest fraction
+    integer :: toosmall_any  ! sum of the landunit thresholds
+    real(r8) :: toosmall(max_lunit)  ! Array of the thresholds (fraction)
+    real(r8) :: residual(max_lunit)  ! Array of wt_lunit residuals (fraction)
+    !-----------------------------------------------------------------------
+
+    SHR_ASSERT_ALL((ubound(wt_lunit) == (/endg, max_lunit/)), errMsg(sourcefile, __LINE__))
+
+    toosmall_any = toosmall_soil + toosmall_crop + toosmall_glacier + &
+                   toosmall_lake + toosmall_wetland + toosmall_urb_tbd + &
+                   toosmall_urb_hd + toosmall_urb_md
+
+    if (toosmall_any > 0) then
```

Comment:
There should also be a check that toosmall_any isn't > 100 right? The namelist puts a limit on that, but that can change, so the code should check the actual issue in case we change the namelist.

Reply:
OK, I realized this was wrong. There isn't an upper limit on toosmall_any. Each of the components can only be up to 100, But, one grid cell could be 100 urban and another 100 soil and that's fine.